### PR TITLE
Switch to use `cf_accessibility_severity` instead of whiteboard tags

### LINKED ISSUE #2069: Use new access severity field in Bugzilla instead of whiteboard tag
We're [migrating](https://bugzilla.mozilla.org/show_bug.cgi?id=1829061) the [access-s] whiteboard tags in Bugzilla to have a dedicated dropdown field. BugBot will need to be updated accordingly.

I'm not quite sure how best to coordinate the timing of this. The Bugzilla migration is ready, but we're holding on executing it. Should the Bugzilla migration wait until the BugBot change lands or should we do it the other way around? I'm not too concerned if we miss a couple of bugs - we can pick those up manually - but it'd be good if we can have these happen within a few days of each other at worst.

Thanks.

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #2069 

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/bugbot/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
This will be ready once we have a response regarding https://mozilla.slack.com/archives/C5F7KFJ4R/p1684539652335299

File: bugbot/rules/duplicate_copy_metadata.py [hunk 1]
```diff
@@ -25,12 +27,24 @@
...
+
+        self.last_modification_date = lmdutils.get_date("today", last_modification_days)
+
     def description(self):
         return "Copied fields from duplicate bugs"
 
     def handle_bug(self, bug, data):
-        bugid = str(bug["id"])
-        data[bugid] = bug
+        if bug["last_change_time_non_bot"] < self.last_modification_date:
```
Comment (marco-c) [inline bugbot/rules/duplicate_copy_metadata.py:44]: This change isn't directly related, right? Could you put it in a separate commit?

File: bugbot/rules/duplicate_copy_metadata.py [hunk 2]
```diff
@@ -15,6 +16,7 @@
     "keywords": "Keywords",
     "severity": "Severity",
     "whiteboard": "Whiteboard",
+    "cf_accessibility_severity": "Accessability Severity",
```
[the remark this answers is not in the document]
Comment (suhaibmujahid) [reply bugbot/rules/duplicate_copy_metadata.py:19]: Fixed in 694c3f33a43fa9a4e42a8b81088804f17bf1b27b

File: bugbot/rules/duplicate_copy_metadata.py (hunk 1 above)
Comment (suhaibmujahid) [reply bugbot/rules/duplicate_copy_metadata.py:44]: It is related to the migration process. The migration script (i.e., https://github.com/mozilla-bteam/bmo/pull/2067) will modify all accessibility bugs. This will ensure we will not pick up old bugs updated by the migration script.

Comment (suhaibmujahid) [reply bugbot/rules/duplicate_copy_metadata.py:44]: I will split it into a separate commit to have a clearer history.

**Review by marco-c — APPROVED** [review]
